src/session/content/message_row/text.rs

```diff
@@ -46,90 +50,96 @@ mod imp {
 
 glib::wrapper! {
     pub struct MessageText(ObjectSubclass<imp::MessageText>)
-        @extends gtk::Widget;
-}
-
-impl Default for MessageText {
-    fn default() -> Self {
-        Self::new()
-    }
+        @extends gtk::Widget, MessageRow;
 }
 
 impl MessageText {
-    pub fn new() -> Self {
-        glib::Object::new(&[]).expect("Failed to create MessageText")
+    pub fn new(message: &glib::Object) -> Self {
+        glib::Object::new(&[("message", message)]).expect("Failed to create MessageText")
     }
 
-    pub fn set_message(&self, message: &Message) {
-        let self_ = imp::MessageText::from_instance(self);
-
-        if message.is_outgoing() {
-            self.add_css_class("outgoing");
-        } else {
-            self.remove_css_class("outgoing");
-        }
-
-        // Show sender label, if needed
-        let show_sender = {
-            if message.is_outgoing() {
-                matches!(message.sender(), MessageSender::Chat(_))
+    fn update_widget(&self) {
+        if let Some(message) = self.message() {
+            let self_ = imp::MessageText::from_instance(self);
+
+            let content_expression = gtk::PropertyExpression::new(
+                Message::static_type(),
+                gtk::NONE_EXPRESSION,
+                "content",
+            );
+            let text_expression = gtk::ClosureExpression::new(
+                move |args| -> String {
+                    let content = args[1].get::<BoxedMessageContent>().unwrap();
+                    format_message_content_text(content.0)
+                },
+                &[content_expression.upcast()],
+            );
+
+            if let Some(message) = message.downcast_ref::<Message>() {
+                // Show sender label, if needed
+                let show_sender = {
+                    if message.is_outgoing() {
+                        matches!(message.sender(), MessageSender::Chat(_))
+                    } else {
+                        matches!(
+                            message.chat().type_(),
+                            ChatType::BasicGroup(_) | ChatType::Supergroup(_)
+                        )
+                    }
+                };
+                if show_sender {
+                    let sender_name_expression = message.sender_name_expression();
+                    sender_name_expression.bind(&*self_.sender_label, "label", gtk::NONE_WIDGET);
+
+                    // Remove the previous color css class
+                    let mut sender_color_class = self_.sender_color_class.borrow_mut();
+                    if let Some(class) = sender_color_class.as_ref() {
+                        self_.sender_label.remove_css_class(class);
```

Comment:
Here we remove the css class. Should this also be done for the sponsored message `else` block?